packages/tools/package/src/commands/changelog/changelog.utils.ts

```diff
@@ -0,0 +1,79 @@
+import { Package } from '../../models';
+import { getCommits } from './changelog.executor';
+import { AlongWithData, ChangelogEntry } from './changelog.types';
+
+const fs = require('fs');
+
+/**
+ * Function to get the along with data
+ * alongWithData object shows the versions of packages that were created with the current package
+ * @param packageName - Name of the package for which we need to create the along with object.
+ * @param packages - Details of all the modified packages.
+ * @returns - AlongWithData for the package.
+ */
+function getAlongWithData(packageName: string, packages: Package[]): AlongWithData {
+  const alongWith: { [key: string]: string } = {};
+  Object.keys(packages).forEach((index: any) => {
+    const pkg = packages[index].name;
+    if (pkg !== packageName) {
+      const exactVersion = packages[index].version;
+      alongWith[pkg] = exactVersion;
+    }
+  });
+  return alongWith;
+}
+
+/**
+ * Function to create or update changelog files
+ * @param packages - Details of the modified packages
+ * @param prevCommitId - commitId of the previous commit
+ */
+export async function createOrUpdateChangelog(packages: Package[], prevCommitId: string) {
+  Object.keys(packages).forEach(async (index: any) => {
+    const pkgName = packages[index].name;
+    // const {version} = packages[index];
+    // console.log(packages[index], version);
+    const { version } = packages[index];
+    const fileName = version.split('-')[0].replace(/\./g, '_');
+    // Constructing the changelog file name
+    const changelogFilePath = `./docs/changelog/v${fileName}.json`;
+    // Prepare the changelog entry
```

Comment:
Let's leave a blank line before (wherever we add a comment)

Reply:
Added.